draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -1482,14 +1500,14 @@ is square by the condition on Z given above.
 Operations:
 
 ~~~
-1. den = inv0(Z^2 * u^4 + Z * u^2)
-2.  x1 = (-B / A) * (1 + den)
-3.  If den == 0, set x1 = B / (Z * A)
+1.  t1 = inv0(Z^2 * u^4 + Z * u^2)
+2.  x1 = (-B / A) * (1 + t1)
+3.  If t1 == 0, set x1 = B / (Z * A)
 4. gx1 = x1^3 + A * x1 + B
 5.  x2 = Z * u^2 * x1
 6. gx2 = x2^3 + A * x2 + B
-7.  If gx1 is square, set x = x1 and y = sqrt(gx1)
-8.  If gx2 is square, set x = x2 and y = sqrt(gx2)
+7.  If is_square(gx1), set x = x1 and y = sqrt(gx1)
+8.  Else if is_square(gx2), set x = x2 and y = sqrt(gx2)
```

Comment:
Since the exceptional cases were handled, there is no need for testing again for `is_square(gx2)`. 

```suggestion
8.  Else, set x = x2 and y = sqrt(gx2)
```